src/pybind/mgr/cephadm/serve.py

```diff
@@ -1074,6 +1075,12 @@ def _create_daemon(self,
                 self.mgr.cache.update_daemon_config_deps(
                     daemon_spec.host, daemon_spec.name(), daemon_spec.deps, start_time)
                 self.mgr.cache.save_host(daemon_spec.host)
+                if redeploy:
+                    self.mgr.check_mon_command({
+                        'prefix': 'config rm',
+                        'name': 'container_image',
+                        'who': utils.name_to_config_section(daemon_spec.daemon_type + '.' + daemon_spec.daemon_id),
+                    })
```

Comment:
Thanks, I see. We want to upgrade mgr only because redeploying osds will restart osds which result in data migration. That is why we exec 'daemon redeploy'. @sebastian-philipp

Reply:
we have to use cases here:
1. user want to redeploy all daemons, but don't want to change anything unexpected
2. user wants to redeploy all daemons of a particular service and want to have all daemons on a single image
3. user wants to redeploy _all_ daemons with a consistent image

how would you differenciate between those use cases?

Reply:
ping?

Reply:
ping?